src/util/sighash.rs

```diff
@@ -229,31 +233,45 @@ impl<'s> ScriptPath<'s> {
     }
 }
 
-impl From<LegacySigHashType> for SigHashType {
-    fn from(s: LegacySigHashType) -> Self {
+impl From<EcdsaSigHashType> for SchnorrSigHashType {
+    fn from(s: EcdsaSigHashType) -> Self {
         match s {
-            LegacySigHashType::All => SigHashType::All,
-            LegacySigHashType::None => SigHashType::None,
-            LegacySigHashType::Single => SigHashType::Single,
-            LegacySigHashType::AllPlusAnyoneCanPay => SigHashType::AllPlusAnyoneCanPay,
-            LegacySigHashType::NonePlusAnyoneCanPay => SigHashType::NonePlusAnyoneCanPay,
-            LegacySigHashType::SinglePlusAnyoneCanPay => SigHashType::SinglePlusAnyoneCanPay,
+            EcdsaSigHashType::All => SchnorrSigHashType::All,
+            EcdsaSigHashType::None => SchnorrSigHashType::None,
+            EcdsaSigHashType::Single => SchnorrSigHashType::Single,
+            EcdsaSigHashType::AllPlusAnyoneCanPay => SchnorrSigHashType::AllPlusAnyoneCanPay,
+            EcdsaSigHashType::NonePlusAnyoneCanPay => SchnorrSigHashType::NonePlusAnyoneCanPay,
+            EcdsaSigHashType::SinglePlusAnyoneCanPay => SchnorrSigHashType::SinglePlusAnyoneCanPay,
         }
     }
 }
 
-impl SigHashType {
+impl SchnorrSigHashType {
     /// Break the sighash flag into the "real" sighash flag and the ANYONECANPAY boolean
-    pub(crate) fn split_anyonecanpay_flag(self) -> (SigHashType, bool) {
+    pub(crate) fn split_anyonecanpay_flag(self) -> (SchnorrSigHashType, bool) {
         match self {
-            SigHashType::Default => (SigHashType::Default, false),
-            SigHashType::All => (SigHashType::All, false),
-            SigHashType::None => (SigHashType::None, false),
-            SigHashType::Single => (SigHashType::Single, false),
-            SigHashType::AllPlusAnyoneCanPay => (SigHashType::All, true),
-            SigHashType::NonePlusAnyoneCanPay => (SigHashType::None, true),
-            SigHashType::SinglePlusAnyoneCanPay => (SigHashType::Single, true),
-            SigHashType::Reserved => (SigHashType::Reserved, false),
+            SchnorrSigHashType::Default => (SchnorrSigHashType::Default, false),
+            SchnorrSigHashType::All => (SchnorrSigHashType::All, false),
+            SchnorrSigHashType::None => (SchnorrSigHashType::None, false),
+            SchnorrSigHashType::Single => (SchnorrSigHashType::Single, false),
+            SchnorrSigHashType::AllPlusAnyoneCanPay => (SchnorrSigHashType::All, true),
+            SchnorrSigHashType::NonePlusAnyoneCanPay => (SchnorrSigHashType::None, true),
+            SchnorrSigHashType::SinglePlusAnyoneCanPay => (SchnorrSigHashType::Single, true),
+            SchnorrSigHashType::Reserved => (SchnorrSigHashType::Reserved, false),
+        }
+    }
+
+    /// Create a [`SchnorrSigHashType`] from raw u8
+    pub fn from_u8(hash_ty: u8) -> Result<Self, Error> {
```

Comment:
Nit: probably we create it from a byte value, not `u8`. I.e. `from_byte` may be more "consensus-style"